### PR TITLE
🐸Restructure Frontend Sources

### PR DESCRIPTION
**Pull request by f0x52**
Include admin and future user panel in main repo, properly deduplicate bundles for css+js across uses

Mostly ready, but needs more testing to prevent regressions, and some documentation updates

### CODE REVIEW

**Comment by f0x52** [comment]
- [ ] documentation
- [ ] check release flow
- [ ] update dev server to work better in tandem with testrig

File: .goreleaser.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ before:
   - sed -i "s/REPLACE_ME/{{ incpatch .Version }}/" docs/api/swagger.yaml
   - cp docs/api/swagger.yaml web/assets/swagger.yaml
   # install and bundle the web assets and styling
```
Comment (f0x52) [inline .goreleaser.yml:13]: goreleaser untested

File: Dockerfile [hunk 2]
```diff
@@ -5,17 +5,18 @@ FROM --platform=${BUILDPLATFORM} node:17.6.0-alpine3.15 AS admin_builder
...
-
-RUN npm install
-RUN node index.js
-
 FROM --platform=${TARGETPLATFORM} alpine:3.15.0 AS executor
 
 # copy over the binary from the first stage
 COPY --chown=1000:1000 gotosocial /gotosocial/gotosocial
 
+# bundle frontend stuff
```
Comment (f0x52) [inline Dockerfile:13]: docker changes are also untested

File: internal/web/base.go [hunk 3]
```diff
@@ -136,16 +136,74 @@ func (m *Module) NotFoundHandler(c *gin.Context) {
 	})
 }
 
+// TODO: abstract the {admin, user}panel handlers in some way
```
Comment (f0x52) [inline internal/web/base.go:139]: good way to abstract this? maybe save it for a later PR

File: web/template/header.tmpl [hunk 4]
```diff
@@ -9,21 +9,23 @@
 	<meta name="og:title" content="GoToSocial Testing Instance">
```
Comment (sexybiggetje) [inline web/template/header.tmpl:9]: The meta og:title should match the title tag here I think?

File: Dockerfile (hunk 2 above)
Comment (tsmethurst) [reply Dockerfile:13]: Unfortunately I think we'll still need to do this as a two-step build where the admin panel is built in the first step (using `--platform=${BUILDPLATFORM}`). Reason for this is that with our current multi-architecture setting, we can't run commands in the second docker container while building.

**Review by tsmethurst — COMMENTED** [review]
Looking good so far :) Since the javascript files from the admin panel etc are now in the main repo, you should probably add the license text that's in the go files to the top of the javascript files as well.

File: Dockerfile (hunk 2 above)
Comment (tsmethurst) [reply Dockerfile:13]: Oh wait scratch that.... I think we can just have GoReleaser do the bundling and then copy the bundled files directly into the Docker container, which would be even neater, then we don't need a two-stage build at all

File: web/template/header.tmpl (hunk 4 above)
Comment (f0x52) [reply web/template/header.tmpl]: yep https://github.com/superseriousbusiness/gotosocial/pull/634/commits/66ec12728ad7c682e44131a8b2c38e2b76ef133b

File: .goreleaser.yml (hunk 1 above)
Comment (tsmethurst) [reply .goreleaser.yml]: tested it :)

File: Dockerfile (hunk 2 above)
Comment (tsmethurst) [reply Dockerfile]: Fixed

File: internal/web/base.go [hunk 5]
```diff
@@ -120,16 +120,74 @@ func (m *Module) baseHandler(c *gin.Context) {
...
+func (m *Module) AdminPanelHandler(c *gin.Context) {
+	l := logrus.WithField("func", "admin-panel")
+	l.Trace("serving admin panel")
+
+	host := config.GetHost()
+	instance, err := m.processor.InstanceGet(c.Request.Context(), host)
+	if err != nil {
+		l.Debugf("error getting instance from processor: %s", err)
```
Comment (tsmethurst) [inline internal/web/base.go:131]: i'll update these new functions to use the new api error handling stuff tomorrow

**Pull request merged**